### PR TITLE
feat: upgrade rc-segmented v2

### PR DESCRIPTION
**Pull request by vagusX**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Since v4.20.0 `Segemented` props `onChange` callback function parameter adjusted from `ChangeEvent` (v4.20.0-alpha.0, v4.20.0-alpha.1) to `value` to simplify API.   |
| 🇨🇳 Chinese |  自 v4.20.0 起 `Segemented` 的 props `onChange` 回调函数的参数从 `ChangeEvent` (v4.20.0-alpha.0、v4.20.0-alpha.1) 调整为 `value`，以简化 API。    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by vagusX** [comment]
https://github.com/react-component/segmented/pull/33

**Comment by vagusX** [comment]
这个要提到 next 嘛 @zombieJ

File: components/segmented/__tests__/index.test.tsx [hunk 1]
```diff
@@ -88,7 +88,7 @@ describe('Segmented', () => {
     const wrapper = mount(
       <Segmented
         options={['Daily', 'Weekly', 'Monthly']}
-        onChange={e => handleValueChange(e.target.value)}
+        onChange={value => handleValueChange(value)}
```
Comment (afc163) [inline components/segmented/__tests__/index.test.tsx:91]:
```suggestion
        onChange={handleValueChange}
```

File: components/segmented/demo/controlled-two.md [hunk 2]
```diff
@@ -22,17 +22,9 @@ const Demo = () => {
   const [foo, setFoo] = useState('AND');
   return (
     <>
-      <Segmented
-        value={foo}
-        options={['AND', 'OR', 'NOT']}
-        onChange={e => setFoo(e.target.value)}
-      />
+      <Segmented value={foo} options={['AND', 'OR', 'NOT']} onChange={value => setFoo(value)} />
```
Comment (afc163) [inline components/segmented/demo/controlled-two.md:25]:
```suggestion
      <Segmented value={foo} options={['AND', 'OR', 'NOT']} onChange={setFoo} />
```

**Comment by afc163** [comment]
图片

写清楚和之前版本（具体哪个版本）的差异。

**Pull request merged**